server/service/src/preference/mod.rs

```diff
@@ -1,33 +1,43 @@
 use crate::service_provider::ServiceContext;
-use preferences::{get_preference_descriptions, get_preferences, Preference, Preferences};
-use repository::{PreferenceRow, RepositoryError};
+
+pub mod types;
+pub use types::*;
+mod query_preference;
 
 pub mod preferences;
+pub use preferences::*;
 pub mod upsert;
+pub mod upsert_helpers;
 
 pub use upsert::*;
 
 pub trait PreferenceServiceTrait: Sync + Send {
-    // Maybe should be called get_store_preferences, but wanting to maintain
-    // distinction from existing store preferences at this stage
-    fn get_preferences(
-        &self,
-        ctx: &ServiceContext,
-        store_id: &str,
-    ) -> Result<Preferences, RepositoryError> {
-        get_preferences(ctx, store_id)
+    fn get_preference_registry(&self) -> PreferenceRegistry {
+        get_preference_registry()
     }
 
-    fn get_preference_descriptions(&self) -> Vec<Box<dyn Preference<Value = bool>>> {
-        get_preference_descriptions()
+    fn get_preference_descriptions(&self, pref_type: PreferenceType) -> Vec<PreferenceDescription> {
+        let PreferenceRegistry {
+            show_contact_tracing,
+        } = &self.get_preference_registry();
```

Comment:
Destructuring here, so should get an error when a new pref is added, to prompt to append to the all prefs below